src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -87,17 +94,17 @@ public static AppDomain CreateDomain(string friendlyName)
             throw new PlatformNotSupportedException();
         }
 
-        public int ExecuteAssembly(string assemblyFile) => ExecuteAssembly(assemblyFile, null, null, Configuration.Assemblies.AssemblyHashAlgorithm.None);
-        public int ExecuteAssembly(string assemblyFile, string[] args) => ExecuteAssembly(assemblyFile, args, null, Configuration.Assemblies.AssemblyHashAlgorithm.None);
-        public int ExecuteAssembly(string assemblyFile, string[] args, byte[] hashValue, Configuration.Assemblies.AssemblyHashAlgorithm hashAlgorithm)
+        public int ExecuteAssembly(string assemblyFile) => ExecuteAssembly(assemblyFile, null);
+        public int ExecuteAssembly(string assemblyFile, string[] args) 
         {
             if (assemblyFile == null)
                 throw new ArgumentNullException(nameof(assemblyFile));
-            /* TODO Assembly.LoadFrom is not implemented yet
-            Assembly assembly = Assembly.LoadFrom(path, hashValue, hashAlgorithm);
+            Assembly assembly = Assembly.LoadFrom(assemblyFile);
             return ExecuteAssembly(assembly, args);
-            */
-            return 0;
+        }
+        public int ExecuteAssembly(string assemblyFile, string[] args, byte[] hashValue, Configuration.Assemblies.AssemblyHashAlgorithm hashAlgorithm)
+        {
+            throw new PlatformNotSupportedException();
```

Comment:
And it has zero usage because of it is only meaningful for very specific partial trust/CAS scenarios - to protect against the case where somebody may be swapping binary files underneath you.